tools/rebase-upstream-commits.ts

```diff
@@ -260,51 +276,86 @@ function continueRebasing() {
     if (unresolvedFiles.match(/^spec\.html/gm)) {
       runInRepository('git rm -f spec.html');
     }
-    const currentTest262CommitHash = getTest262Commit('HEAD', 'test262');
-    const upstreamTest262CommitHash = getTest262Commit(currentConflict, 'polyfill/test262');
-    if (currentTest262CommitHash && upstreamTest262CommitHash) {
-      const updated262Commit = getNewestCommit(
-        currentTest262CommitHash,
-        upstreamTest262CommitHash,
-        currentExecOpts.repository + '/test262'
-      );
-      // const upstreamTest262CommitRaw = String(runInRepository(`git
-      // ls-tree ${currentConflict} polyfill/test262`)); if
-      // (!upstreamTest262CommitRaw) {
-      //   throw new Error(`Unable to find Test262 commit for upstream
-      //   ${currentConflict}`);
-      // }
-      // const upstreamTest262CommitHash =
-      // upstreamTest262CommitRaw.split(/\s+/)[2];
-      if (updated262Commit !== currentTest262CommitHash) {
-        runInRepository(`(cd test262 && git checkout ${updated262Commit})`);
+    const filesInUpstreamCommit = runInRepository(
+      `git diff-tree --name-only --no-commit-id -r ${currentConflict}`
+    ).split('\n');
+    if (filesInUpstreamCommit.includes('polyfill/test262')) {
+      const currentTest262CommitHash = getTest262Commit('HEAD', 'test262');
+      const upstreamTest262CommitHash = getTest262Commit(currentConflict, 'polyfill/test262');
+      if (currentTest262CommitHash && upstreamTest262CommitHash) {
+        const updated262Commit = getNewestCommit(
+          currentTest262CommitHash,
+          upstreamTest262CommitHash,
+          currentExecOpts.repository + '/test262'
+        );
+        if (updated262Commit !== currentTest262CommitHash) {
+          runInRepository(`(cd test262 && git checkout ${updated262Commit})`);
+        }
+      }
+      try {
+        runInRepository('git rm -rf polyfill/test262');
+      } catch (e) {
+        // Do nothing - this path might not need resolving according to git
+        // in this commit.
       }
     }
-    try {
-      runInRepository('git rm -rf polyfill/test262');
-    } catch (e) {
-      // Do nothing - this path might not need resolving in this commit.
+
+    if (filesInUpstreamCommit.includes('polyfill/lib/ecmascript.mjs')) {
+      log(`
+Changes to the ecmascript.mjs file in the upstream repository are never
+automatically rebased onto lib/ecmascript.ts, because the files don't
+share history and git doesn't think they are at all similar.
+
+To resolve this conflict you will need to manually inspect and re-apply
+changes from polyfill/lib/ecmascript.mjs over to lib/ecmascript.ts. Once
+done, remove polyfill/lib/ecmascript.mjs (using git rm) and add the new
+changes to lib/ecmascript.ts (using git add).
```

Comment:
Might be nice to add copy/pasteable command lines here.

```suggestion
done, remove the mjs file from the commit using:
  git rm polyfill/lib/ecmascript.mjs
Then stage the TS version of that file using:
  git add lib/ecmascript.ts 
At that point, after any other changes are staged too, you can:
  trt continue
```